src/messaging/tests/MessagingContext.h

```diff
@@ -92,16 +93,12 @@ class MessagingContext : public PlatformMemoryUser
     static constexpr System::Clock::Timeout kResponsiveIdleRetransTimeout   = System::Clock::Milliseconds32(10);
     static constexpr System::Clock::Timeout kResponsiveActiveRetransTimeout = System::Clock::Milliseconds32(10);
 
-    MessagingContext() :
-        mInitialized(false), mAliceAddress(Transport::PeerAddress::UDP(GetAddress(), CHIP_PORT + 1)),
-        mBobAddress(LoopbackTransport::LoopbackPeer(mAliceAddress))
-    {}
-    // TODO Replace VerifyOrDie with Pigweed assert after transition app/tests to Pigweed.
-    // TODO Currently src/app/icd/server/tests is using MessagingConetext as dependency.
-    ~MessagingContext() { VerifyOrDie(mInitialized == false); }
+    MessagingContext() : mpData(new MessagingContextData()) {}
```

Comment:
since you new/delete automatically, would it make sense to use a std::unique_ptr so that we do not need to track down lifetime?

Reply:
Why is mpData a pointer rather than just a member if it is allocated in new and freed up in the destructor? Is it stack space saving (use heap for large objects instead of stack)? If yes, this should be in a comment somewhere.

Reply:
I replaced this with std:unique_ptr.

Yup, the reason for making mpData a pointer is to put it on the heap and save stack space.  I have added a comment to explain this.